### PR TITLE
portable: add GetImageStateWithExtensions method

### PR DESCRIPTION
**Pull request by bluca**
Allow to correctly query a layered portable service for
attached/detached state.

### CODE REVIEW

File: src/portable/portabled-bus.c [hunk 1]
```diff
@@ -235,9 +237,28 @@ static int method_get_image_state(sd_bus_message *message, void *userdata, sd_bu
...
+                r = sd_bus_message_read_strv(message, &extension_images);
+                if (r < 0)
+                        return r;
+
+                r = sd_bus_message_read(message, "t", &input_flags);
+                if (r < 0)
+                        return r;
+
+                /* No flags are supported by this method for now. */
+                if (input_flags!= 0)
```
Comment (yuwata) [inline src/portable/portabled-bus.c]: missing white space.

File: src/portable/portablectl.c [hunk 2]
```diff
@@ -1049,9 +1049,29 @@ static int is_image_attached(int argc, char *argv[], void *userdata) {
...
+
+        r = bus_message_new_method_call(bus, &m, bus_portable_mgr, method);
+        if (r < 0)
+                return bus_log_create_error(r);
+
+        r = sd_bus_message_append(m, "s", image);
+        if (r < 0)
+                return bus_log_create_error(r);
+
+        r = attach_extensions_to_message(m, arg_extension_images);
```
Comment (yuwata) [inline src/portable/portablectl.c:1062]: Is this OK to be called for `GetImageState`?

**Review by yuwata — CHANGES_REQUESTED** [review]

File: src/portable/portablectl.c (hunk 2 above)
Comment (bluca) [reply src/portable/portablectl.c:1062]: yes, this helper is made to make that simpler, it will do nothing if the strv is empty

Comment (yuwata) [reply src/portable/portablectl.c:1062]: Thanks.

Comment (yuwata) [reply src/portable/portablectl.c:1062]: Hm, it calls `sd_bus_message_open_container(m, 'a', "s"); sd_bus_message_close_container(m);` even if `arg_extension_images` is empty. I cannot remember right now and not sure, is it really does nothing to the message?

I think it is simpler to move this to the if block below.

Comment (bluca) [reply src/portable/portablectl.c:1062]: It returns immediately as the first thing if the strv is empty:

https://github.com/systemd/systemd/blob/f338a496e48392929400916ebb0eb9819dec1e2b/src/portable/portablectl.c#L98

Comment (yuwata) [reply src/portable/portablectl.c:1062]: Ouch. Indeed.

**Review by yuwata — APPROVED** [review]

**Pull request merged**